lib/driver.js

```diff
@@ -232,14 +232,24 @@ class XCUITestDriver extends BaseDriver {
     await this.runReset();
 
     log.info(`Setting up ${this.isRealDevice() ? 'real device' : 'simulator'}`);
-    if (!this.isRealDevice()) {
+
+    if (this.isRealDevice()) {
+      if (this.opts.app) {
+        await this.installApp();
+      }
+    } else {
+      const installAppPromise = new Promise((resolve) => {
+        this.opts.device.on(BOOT_COMPLETED_EVENT, async () => {
```

Comment:
this anonymous `async` function is a source of unhandled rejections (what if `this.installApp()` fails?), since nothing is `await`ing it. To write it  safely, it should look like this:

```js
const installAppPromise = new Promise((resolve) => {
  this.opts.device.on(BOOT_COMPLETED_EVENT, async () => {
    try {
      if (this.opts.app) {
        await this.installApp();
      }
      resolve();
    } catch (e) {
      reject(e);
    }
  });
});
```

have to be careful when mixing promise styles in this way :-)